ethcore/src/state/mod.rs

```diff
@@ -310,6 +311,11 @@ impl<B: Backend> State<B> {
 		Ok(state)
 	}
 
+	/// Get a VM factory that can execute on this state.
+	pub fn vm_factory(&self) -> EvmFactory {
```

Comment:
I did not move it in the state, it is there already. The state is currently constructed with all the `Factories` that are going to operate on it. I would agree that the state should not be concerned about factories, but refactoring this is out of the scope of this PR.